### PR TITLE
[Impeller] fix colr/bitmap font color drawing.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/126546

Track the paint color used by Bitmap/Emoji/COLR fonts and use it as a cache key. This allows non-COLR glyphs in a COLR font to get the correct text color. For other kinds of fonts, we always record the color as black so there are no cache efficiency hits in general.

### CODE REVIEW

File: impeller/geometry/color.h [hunk 1]
```diff
@@ -253,6 +253,11 @@ struct Color {
     return {r, g, b, a};
   }
 
+  constexpr uint32_t ToSkia() const {
```
Comment (jonahwilliams) [inline impeller/geometry/color.h:256]: I should probably call this ToARGB

File: impeller/typographer/text_frame.h [hunk 2]
```diff
@@ -60,6 +63,8 @@ class TextFrame {
   ///             to apply opacity peephole optimizations to text blobs.
   bool MaybeHasOverlapping() const;
 
+  Color GetColor() const;
```
Comment (jonahwilliams) [inline impeller/typographer/text_frame.h:66]: need doc comment.

File: impeller/geometry/color.h (hunk 1 above)
Comment (jonahwilliams) [reply impeller/geometry/color.h]: Done

**Review by matanlurey — APPROVED** [review]
Seems good to me. Nice tests and doc comments!

File: impeller/typographer/font_glyph_pair.h [hunk 3]
```diff
@@ -40,15 +41,17 @@ struct FontGlyphPair {
 template <>
 struct std::hash<impeller::ScaledFont> {
   constexpr std::size_t operator()(const impeller::ScaledFont& sf) const {
-    return fml::HashCombine(sf.font.GetHash(), sf.scale);
+    return fml::HashCombine(sf.font.GetHash(), sf.scale, sf.color.red,
```
Comment (jason-simmons) [inline impeller/typographer/font_glyph_pair.h:44]: Can simplify by calling `sf.color.ToARGB()` here

**Review by jason-simmons — APPROVED** [review]

File: impeller/typographer/font_glyph_pair.h (hunk 3 above)
Comment (jonahwilliams) [reply impeller/typographer/font_glyph_pair.h:44]: Ahh good idea!

**Pull request merged**